### PR TITLE
Add Support for HKEYS command. 

### LINKED ISSUE #649: Add support for command `HKEYS`
Add support for the `HKEYS` command in DiceDB similar to the [`HKEYS` command in Redis](https://redis.io/docs/latest/commands/command-list/). Please refer to the following commit in Redis to understand the implementation specifics - [source](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa).

Write unit and integration tests for the command referring to the tests written in the [Redis codebase 7.2.5](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa). For integration tests, you can refer to the `tests` folder. Note: they have used `TCL` for the test suite, and we need to port that to our way of writing integration tests using the relevant helper methods. Please refer to our [tests directory](https://github.com/DiceDB/dice/tree/master/tests).

For the command, benchmark the code and measure the time taken and memory allocs using [`benchmem`](https://blog.logrocket.com/benchmarking-golang-improve-function-performance/) and try to keep them to the bare minimum.

### PR DESCRIPTION
**Pull request by swarajrb7**
Screenshot 2024-10-01 005712

image

Fixes #649

### CODE REVIEW

File: internal/eval/commands.go [hunk 1]
```diff
@@ -620,6 +620,13 @@ var (
 		Arity:    -4,
 		KeySpecs: KeySpecs{BeginIndex: 1},
 	}
+	hkeysCmdMeta = DiceCmdMeta{
+		Name: "HKEYS",
+		Info:  `HKEYS command is used to retrieve all the keys(or field names) within a hash. Complexity is O(n) where n is the size of the hash.`,
+		Eval: evalHKEYS,
+		Arity: -1,
```
Comment (apoorvyadav1111) [inline internal/eval/commands.go:627]: Since HKEYS only accepts key as the argument, we can keep Arity as 1.

File: internal/eval/eval.go [hunk 2]
```diff
@@ -3030,6 +3030,33 @@ func evalHSET(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(numKeys, false)
 }
```
Comment (apoorvyadav1111) [inline internal/eval/eval.go:3032]: Could you please add 2-3 lines of comment describing the function on a higher level.
Thanks

**Review by apoorvyadav1111 — CHANGES_REQUESTED** [review]
Hey, thanks for the changes, left some comments.

File: internal/eval/commands.go (hunk 1 above)
Comment (swarajrb7) [reply internal/eval/commands.go:627]: it was a typo will fix it.

**Comment by swarajrb7** [comment]
@apoorvyadav1111 i have made the required changes 
Please take a look.

**Review by apoorvyadav1111 — APPROVED** [review]

**Review by lucifercr07 — APPROVED** [review]

**Pull request merged**